### DISCUSSION TITLE
Worledit

### DISCUSSION THREAD

**Discussion by pizzav-xyz** [maruohon/litematica#849] [Q&A]
How do I make circle selection like worldedit and save it

**Comment by maruohon** [comment]
All selections and schematics in Litematica are rectangular boxes. So you'll just have to make a selection box around the circle.

**Reply by pizzav-xyz** [reply]
Can you make something like this

**Reply by maruohon** [reply]
Non-rectangular selections would be a lot more complicated for very little benefit IMO. If you have a non-rectangular area then you can't iterate through it with a simple 3-level for-loop anymore, you need something totally different. And it would also change the schematic format. So that will not happen.

Just make a box selection around the circle shape. If there is other stuff within the box that you don't want in the schematic, then you can use the Schematic Edit mode to remove those after saving the first version.

There will probably be some additional features to help with this kind of stuff in some (distant) future versions, but non-rectangular schematics are almost definitely never going to happen. Circular or sphere selections among other things *might* get added, but if those would be usable for saving schematics at all, then the resulting schematic will still be essentially rectangular in the end, the non-rectangular selection would only allow you to exclude stuff while saving it.

**Reply by pizzav-xyz** [reply]
just like its an option so it doesnt change iterate and it wont change the schematic format and i dont have a circle shape and i just want to to use it for the fill thing

**Reply by rycbar0** [reply]
if you want a perfect circle schematic there are other options for that. world edit for example. i belive you are also in the baritone discord. baritone can also make circles with the #selection command.
one of masa's other mods can also make perfect circles if im not mistaken but i dont remember the name.
if you need a schematic of a circular structure do as masa said and just make the box large enough so that the structure fits inside or if you need something more form fittig make a lot of subregions to only cover the the structure while ignoring the outside. the subregions get combined in 1 schematic but its a lot of effort to configure all the subregions for no real gain.

**Reply by pizzav-xyz** [reply]
> if you want a perfect circle schematic there are other options for that. world edit for example. i belive you are also in the baritone discord. baritone can also make circles with the #selection command. one of masa's other mods can also make perfect circles if im not mistaken but i dont remember the name. if you need a schematic of a circular structure do as masa said and just make the box large enough so that the structure fits inside or if you need something more form fittig make a lot of subregions to only cover the the structure while ignoring the outside. the subregions get combined in 1 schematic but its a lot of effort to configure all the subregions for no real gain.

baritone just fills the slection instantly i want to make a selection manipluate it and build it after
